ckanext/dia/views.py

```diff
@@ -75,3 +81,43 @@ def list():
 
     extra_vars = {u'page': page, u'q': q}
     return tk.render(u'uris/list.html', extra_vars)
+
+@uri_minter.route('/uri/<int:uri_id>/', methods=['GET', 'POST'])
+def edit_uri(uri_id):
+    current_uri = MintedURI.get(uri_id)
+    if not current_uri:
+        base.abort(404, 'URI not found')
+    if current_uri.superseded_by != None:
+        base.aport(422, 'URI is archived, editing not supported')
+
+    data = {'type': current_uri.type, 'name': current_uri.name}
+    vars = {'data': data, 'errors': None, 'error_summary': None, 'update': True}
+    if request.method == 'POST':
+        data_dict = dict(request.form)
+        data_dict['type'] = current_uri.type
+        # checkbox field name is present in request.form if
+        # checked, missing if not checked
+        if 'regenerate' in data_dict:
+            data_dict['regenerate'] = True
```

Comment:
Explicitly set to true or false this way.

```suggestion
        data_dict['regenerate'] = 'regenerate' in data_dict
```